tests/models/test_transformers.py

```diff
@@ -1,18 +1,59 @@
-import guidance
-from guidance import gen, capture
+from textwrap import dedent
+
+import pytest
+
+from guidance import gen, select
+
 from ..utils import get_model
 
+
 def test_gpt2():
     gpt2 = get_model("transformers:gpt2")
     lm = gpt2 + "this is a test" + gen("test", max_tokens=10)
     assert len(str(lm)) > len("this is a test")
 
+
 def test_recursion_error():
-    '''This checks for an infinite recursion error resulting from a terminal node at the root of a grammar.'''
+    """This checks for an infinite recursion error resulting from a terminal node at the root of a grammar."""
     gpt2 = get_model("transformers:gpt2")
 
     # define a guidance program that adapts a proverb
-    lm = gpt2 + f"""Tweak this proverb to apply to model instructions instead.
+    lm = (
+        gpt2
+        + f"""Tweak this proverb to apply to model instructions instead.
     {gen('verse', max_tokens=2)}
     """
-    assert len(str(lm)) > len("Tweak this proverb to apply to model instructions instead.\n\n")
\ No newline at end of file
+    )
+    assert len(str(lm)) > len(
+        "Tweak this proverb to apply to model instructions instead.\n\n"
+    )
+
+
+TRANSFORMER_MODELS = ["gpt2", "microsoft/phi-2"]
```

Comment:
not for this PR, but we may want to explore different config options tied to each model in a dictionary somewhere in the future. e.g. phi-2 needs to be initialized with `trust_remote_code`, and other HF models have their own quirks too. just a note to keep in mind for later, no need to squeeze in as part of this

Reply:
Agreed.